pkg/external/external.go

```diff
@@ -31,13 +33,46 @@ type Event struct {
 	Args                []Argument `json:"args"` //Arguments are ordered according their appearance in the original event
 }
 
+const (
+	TraceeEventSrc         = "tracee-ebpf"
+	TraceeEventContentType = "tracee.event"
+)
+
+func ToEvent(e TraceeEvent) types.Event {
+	origin := TraceeEventSrc + "/" + string(e.EventOrigin())
+	eType := TraceeEventContentType + "." + e.EventName
+
+	return types.Event{
+		Header: types.EventHeader{
+			Type:   eType,
+			Origin: origin,
+		},
+		Payload: e,
+	}
+}
+
+type TraceeEventOrigin string
+
+const (
+	ContainerOrigin TraceeEventOrigin = "container"
+	HostOrigin      TraceeEventOrigin = "host"
+)
+
+func (e TraceeEvent) EventOrigin() TraceeEventOrigin {
+	if e.ContainerID != "" || e.ProcessID != e.HostProcessID {
+		return ContainerOrigin
+	} else {
+		return HostOrigin
+	}
+}
+
 // ToUnstructured returns a JSON compatible map with string, float, int, bool,
 // []interface{}, or map[string]interface{} children.
 //
 // It allows this Event to be manipulated generically. For example, it can be
 // used as a parsed input with OPA SDK to avoid relatively expensive JSON
 // encoding round trip.
-func (e Event) ToUnstructured() (map[string]interface{}, error) {
+func (e TraceeEvent) ToUnstructured() (map[string]interface{}, error) {
```

Comment:
I think we can safely remove ToUnstructured as it's no longer used by the rules engine (see ToParsedEvent method in engine.go) Anyway this method doesn't make much sense anymore. We introduced it some time back hoping to improve performance of OPA evaluations.

Reply:
Done